compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolEnter.java

```diff
@@ -1166,14 +1166,12 @@ public void initPredeclaredModules(Map<Name, BPackageSymbol> predeclaredModules,
 
     @Override
     public void visit(BLangXMLNS xmlnsNode) {
-        String nsURI;
+        String nsURI = "";
         if (xmlnsNode.namespaceURI.getKind() == NodeKind.SIMPLE_VARIABLE_REF) {
             BLangSimpleVarRef varRef = (BLangSimpleVarRef) xmlnsNode.namespaceURI;
-            if (missingNodesHelper.isMissingNode(varRef.variableName.value)) {
-                nsURI = "";
-            } else {
-                // TODO: handle const-ref (#24911)
-                nsURI = "";
+            if (!missingNodesHelper.isMissingNode(varRef.variableName.value) &&
```

Comment:
Why do we require this check `!missingNodesHelper.isMissingNode(varRef.variableName.value)`? I believe we may no longer need this check.

Reply:
Fixed

Reply:
This check can still evaluate to true and generally help with accuracy, although probably not specifically with the const symbol check.

```ballerina
xmlns as ns0;
```

Reply:
If a node is missing, we are issuing an error in the parser. What is the reason for rechecking this on the compiler side?

Reply:
Compilation will still continue when possible. As it does with the snippet I've shared.

Reply:
We can skip this condition here right. Since we are by default setting the value here to "".